src/System.Numerics.Vectors/src/System/Numerics/Matrix4x4.cs

```diff
@@ -896,11 +896,11 @@ public static Matrix4x4 CreatePerspectiveFieldOfView(float fieldOfView, float as
             result.M21 = result.M23 = result.M24 = 0.0f;
 
             result.M31 = result.M32 = 0.0f;
-            result.M33 = farPlaneDistance / (nearPlaneDistance - farPlaneDistance);
+            result.M33 = float.IsPositiveInfinity(farPlaneDistance) ? -1.0f : farPlaneDistance / (nearPlaneDistance - farPlaneDistance);
```

Comment:
I just assumed that since the line above said `result.M31 = result.M32 = 0.0f;` reading struct fields back out was OK. 
If you can suggest a good name for a local I'll be happy to introduce one.